### ISSUE TITLE
Separate Language for site and admin

### ISSUE THREAD

**Issue by aghili371** [OrchardCMS/OrchardCore#3665]
i think it's better to have other option in setting to select admin language.

**Comment by Skrypt** [comment]
You should already be able to do that unless you mean the Cookie provider will take precedence on your admin site if you have set it previously from the frontend website. Is it just a thought or you have an actual issue with this?

**Comment by aghili371** [comment]
@Skrypt I mean one option different than the culture of the web site, just for UI and translate of the admin panel.

**Comment by Skrypt** [comment]
Yes, it's an admin culture picker. We should have this.

**Comment by hishamco** [comment]
@Skrypt if even we have it, one of the culture pickers will reflect on the other because both are using the same cookie. What I suggest here is to add a culture field for user preferences, that way you will not enforce all who using the admin site, so everyone will be happy with his/her favorite language 😄

**Comment by hishamco** [comment]
@Skrypt please tag all the localization related issues with `localization` tag, so we can easily track and fixed them

**Comment by sebastienros** [comment]
We should do a custom feature that would add a culture picker in the admin, and which would set it for the admin only.

**Comment by hishamco** [comment]
I think this will change the default culture for the front-end site, while we already have the ability to do this from Site Settings -> Default Culture

> and which would set it for the admin only

Can your please elaborate

**Comment by aghili371** [comment]
> which would set it for the admin only.

it could be in user fields as new property and set for all any user in the other role.

**Comment by hishamco** [comment]
This is need time to think about it from many perspectives, but while It's go to backlog that mean it's not in 1.0.0

**Comment by sebastienros** [comment]
I think we can have it, it's just that it won't block RC for sure. I am not very consistent when triaging, sorry. I agree with the feature, a back-end user setting could define the preferred culture for the admin. And that would only redefine the CurrentCulture for the Admin. It would be a custom RequestCultureProvider that would use this settings, and would have a higher priority.

**Comment by hishamco** [comment]
I saw some websites that require users to choose language in the login screen, we can do something similar or add it in the user profile if we do so

**Comment by sebastienros** [comment]
This could be a new "Default admin culture" separate from the "Default culture" which is also used for the content localization.
And also a user based setting, that would override the "default admin culture". I think it's more of a database value, that would flow across browsers instead of a local storage one.

We need to ensure that the User Settings page will hide the settings that the user can't change, for instance Roles/IsEnabled based on permissions. This way a user can change its own settings, and admins too. Including preferences (timezone, culture, ...).
We might also want to add dynamic permissions for "Custom User Settings" when it's implemented.

Note for myself: we talked about splitting front-end/back-end user stores. We actually already have solved the distinction with the "Access Admin Panel" permission.

**Comment by dodyg** [comment]
What is the best way to deal with this at the moment? All my websites are LTR/RTL and it's really jarring to see the admin pages switches side based on the public page culture.

**Comment by Skrypt** [comment]
And also a user based setting, that would override the "default admin culture".
You set the CookieRequestCultureProvider with it.

**Comment by hishamco** [comment]
Regarding `CookieRequestCultureProvider` you need another cookie name to avoid a collision with the default one that has been used by `LocalizationMiddleware`

**Comment by Skrypt** [comment]
Yeah don't make a mistake thinking that the wysiwyg editors should not switch based on culture. So, if you want to see your admin UI in "en-ca" while editing a content item set to a RTL culture the wysiwyg editor needs to still be RTL while the page is in "en". There is probably something more to be done than just adding another Cookie there.

**Comment by hishamco** [comment]
> There is probably something more to be done than just adding another Cookie there.

Exactly, I may work on this in `OrchardContrib.Modules` if I have a time

**Comment by jagbarcelo** [comment]
I just wanted to throw my two cents on this idea with someting else none of you mentioned. Every http request from the client provides the `Accept-Language` header. Such as, for instance, `Accept-Language: es,es-ES;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6`

Would it be too difficult for the admin pages just to ignore the cookie set at the public website about language and localization
`AspNetCore.Culture=c%3Des-ES%7Cuic%3Des-ES   -> c=es-ES|uic=es-ES`
and just try tu switch to any of the languages supported by the browser, at least as a failback or alternative method.

**Comment by Skrypt** [comment]
Yes, having a separate cookie for admin and frontend is part of the solution. Though, in the admin when we edit/create a content item and we pick a culture from the LocalizationPart, some elements like the Wysiwyg editors needs to take the LocalizationPart.Culture in consideration. So, even if we set a different cookie, here we have exceptions because of the LocalizationPart.

**Comment by hishamco** [comment]
We already have `AdminCulturePicker`
